Cargo.toml

```diff
@@ -0,0 +1,5 @@
+[workspace]
+members = ["linkerd-network-validator"]
```

Comment:
Not a blocker, but... I would probably leave the directory as `validator` or `network-validator` and then have the crate name be `linkerd-network-validator` -- this how we handle things in the linkerd2 repo, where the directory is `policy-controller` but the crate name is `linkerd-policy-controller`.

Basically, the `linkerd` prefix can be implied from the repo. But we want `linkerd-` in the application's name so that we can use a log setting like `linkerd=debug`, etc`. (And in general, when the binary is produced, it's better for it to be clearly marked as part of linkerd).